test/unit/controllers/consolidations/AssociateDucrSummaryControllerSpec.scala

```diff
@@ -73,131 +62,102 @@ class AssociateDucrSummaryControllerSpec extends ControllerSpec with MockSubmiss
-          .thenReturn(Future.successful(mock[HttpResponse]))
+        assertThrows[IncompleteApplication] {
+          await(controller.displayPage()(getRequest()))
+        }
+      }
 
-        val result = controller.submit()(postRequest(Json.obj()))
+      "Associate Ducr is missing during displaying page" in {
 
-        status(result) mustBe SEE_OTHER
-      }
+        withCaching(MucrOptions.formId, Some(MucrOptions("MUCR")))
+        withCaching(AssociateDucr.formId, None)
 
-      "call SubmissionService and CustomsCacheService in correct order" in {
-
-        when(mockSubmissionService.submitDucrAssociation(any(), any())(any(), any()))
-          .thenReturn(Future.successful(ACCEPTED))
-        when(mockCustomsCacheService.remove(anyString())(any(), any()))
-          .thenReturn(Future.successful(mock[HttpResponse]))
-
-        controller.submit()(postRequest(Json.obj())).futureValue
-
-        val inOrder: InOrder = Mockito.inOrder(mockSubmissionService, mockCustomsCacheService)
-        inOrder
-          .verify(mockCustomsCacheService)
-          .fetchAndGetEntry[MucrOptions](any(), meq(MucrOptions.formId))(any(), any(), any())
-        inOrder
-          .verify(mockCustomsCacheService)
-          .fetchAndGetEntry[AssociateDucr](any(), meq(AssociateDucr.formId))(any(), any(), any())
-        inOrder
-          .verify(mockSubmissionService)
-          .submitDucrAssociation(meq(MucrOptions("MUCR")), meq(AssociateDucr("DUCR")))(any(), any())
-        inOrder.verify(mockCustomsCacheService).remove(anyString)(any(), any())
+        assertThrows[IncompleteApplication] {
+          await(controller.displayPage()(getRequest()))
+        }
       }
 
-      "Redirect to next page" in {
-
-        when(mockSubmissionService.submitDucrAssociation(any(), any())(any(), any()))
-          .thenReturn(Future.successful(ACCEPTED))
-        when(mockCustomsCacheService.remove(anyString())(any(), any()))
-          .thenReturn(Future.successful(mock[HttpResponse]))
+      "Mucr Options is missing during submitting page" in {
 
-        val result = controller.submit()(postRequest(Json.obj()))
+        withCaching(MucrOptions.formId, None)
 
-        redirectLocation(result) mustBe Some(routes.AssociateDucrConfirmationController.displayPage().url)
+        assertThrows[IncompleteApplication] {
+          await(controller.submit()(postRequest(Json.obj())))
+        }
       }
 
-      "add MUCR to Flash" in {
-
-        when(mockSubmissionService.submitDucrAssociation(any(), any())(any(), any()))
-          .thenReturn(Future.successful(ACCEPTED))
-        when(mockCustomsCacheService.remove(anyString())(any(), any()))
-          .thenReturn(Future.successful(mock[HttpResponse]))
+      "Associate Ducr is missing during submitting page" in {
 
-        val result = controller.submit()(postRequest(Json.obj()))
+        withCaching(MucrOptions.formId, Some(MucrOptions("MUCR")))
```

Comment:
Is this line necessary for test to run correctly?

Reply:
Yes, the same explanation like above